src/conditions/datadog_search.rs

```diff
@@ -0,0 +1,658 @@
+            (
+                "-a:[1 TO 10]",
+                log_event!["tags" => vec!["a:2"]],
+                log_event!["tags" => vec!["a:1"]],
+            ),
+            // Range - numeric, inclusive, unbounded (upper), tag.
+            (
+                "a:[50 TO *]",
+                log_event!["tags" => vec!["a:6"]],
+                log_event!["tags" => vec!["a:40"]],
+            ),
+            // Range - numeric, inclusive, unbounded (upper), tag (negate).
+            (
+                "NOT a:[50 TO *]",
+                log_event!["tags" => vec!["a:40"]],
+                log_event!["tags" => vec!["a:6"]],
+            ),
+            // Range - numeric, inclusive, unbounded (upper), tag (negate w/-).
+            (
+                "-a:[50 TO *]",
+                log_event!["tags" => vec!["a:40"]],
+                log_event!["tags" => vec!["a:6"]],
+            ),
+            // Range - numeric, inclusive, unbounded (lower), tag.
+            (
+                "a:[* TO 50]",
+                log_event!["tags" => vec!["a:400"]],
+                log_event!["tags" => vec!["a:600"]],
+            ),
+            // Range - numeric, inclusive, unbounded (lower), tag (negate).
+            (
+                "NOT a:[* TO 50]",
+                log_event!["tags" => vec!["a:600"]],
+                log_event!["tags" => vec!["a:400"]],
+            ),
+            // Range - numeric, inclusive, unbounded (lower), tag (negate w/-).
+            (
+                "-a:[* TO 50]",
+                log_event!["tags" => vec!["a:600"]],
+                log_event!["tags" => vec!["a:400"]],
+            ),
+            // Range - numeric, inclusive, unbounded (both), tag.
+            (
+                "a:[* TO *]",
+                log_event!["tags" => vec!["a:test"]],
+                log_event!["tags" => vec!["b:test"]],
+            ),
+            // Range - numeric, inclusive, unbounded (both), tag (negate).
+            (
+                "NOT a:[* TO *]",
+                log_event!["tags" => vec!["b:test"]],
+                log_event!["tags" => vec!["a:test"]],
+            ),
+            // Range - numeric, inclusive, unbounded (both), tag (negate w/-).
+            (
+                "-a:[* TO *]",
+                log_event!["tags" => vec!["b:test"]],
+                log_event!["tags" => vec!["a:test"]],
+            ),
+            // Range - numeric, inclusive, facet.
+            (
+                "@b:[1 TO 10]",
+                log_event!["custom" => json!({"b": 5})],
+                log_event!["custom" => json!({"b": 11})],
+            ),
+            // Range - numeric, inclusive, facet (negate).
+            (
+                "NOT @b:[1 TO 10]",
+                log_event!["custom" => json!({"b": 11})],
+                log_event!["custom" => json!({"b": 5})],
+            ),
+            // Range - numeric, inclusive, facet (negate w/-).
+            (
+                "-@b:[1 TO 10]",
```

Comment:
As we discussed offline, here we should also make sure that for string values we do alphanumerical comparisons.